Spigot-Server-Patches/0164-Fix-FallingBlocks-being-stuck-on-fences.patch

```diff
@@ -0,0 +1,68 @@
+From 25a066dad9f22b34ad014b712ce931019e896116 Mon Sep 17 00:00:00 2001
+From: Martin Panzer <postremus1996@googlemail.com>
+Date: Fri, 3 Jun 2016 23:13:39 +0200
+Subject: [PATCH] Fix FallingBlocks being stuck on fences
+
+Fallingblocks would previously only check, if directly beneath them a block exists. They also
+hover on top of the 1.5 block tall hitbox of fences during these check. This
+resulted in them always thinking they would be on air.
+
+We now first check, if there if we are already on the ground.
+if not, we check the hitbox of the block at y - 1, using the hitboxes of the blocks.
+
+diff --git a/src/main/java/net/minecraft/server/EntityFallingBlock.java b/src/main/java/net/minecraft/server/EntityFallingBlock.java
+index a5d8f02..cb5a045 100644
+--- a/src/main/java/net/minecraft/server/EntityFallingBlock.java
++++ b/src/main/java/net/minecraft/server/EntityFallingBlock.java
+@@ -2,7 +2,9 @@ package net.minecraft.server;
+ 
+ import com.google.common.collect.Lists;
+ import java.util.ArrayList;
++import java.util.Arrays;
+ import java.util.Iterator;
++import java.util.List;
+ import javax.annotation.Nullable;
+ 
+ import org.bukkit.craftbukkit.event.CraftEventFactory; // CraftBukkit
+@@ -95,10 +97,9 @@ public class EntityFallingBlock extends Entity {
+                 blockposition = new BlockPosition(this);
+                 if (this.onGround) {
+                     IBlockData iblockdata = this.world.getType(blockposition);
+-
+-                    if (BlockFalling.i(this.world.getType(new BlockPosition(this.locX, this.locY - 0.009999999776482582D, this.locZ)))) {
++                    if (!isOnGround()) {
+                         this.onGround = false;
+-                        // return; // CraftBukkit
++                        return; // Paper
+                     }
+ 
+                     this.motX *= 0.699999988079071D;
+@@ -155,6 +156,25 @@ public class EntityFallingBlock extends Entity {
+         }
+     }
+ 
++    // Paper start
++    private boolean isOnGround() {
++        BlockPosition where = new BlockPosition(this.locX, this.locY - 0.009999999776482582D, this.locZ);
++
++        if (!BlockFalling.i(this.world.getType(where))) {
```

Comment:
Should have an obfhelper. ``canGoThrough`` or ``canMoveThrough`` or something similar.